install/kubernetes/helm/istio/templates/service.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "istio.fullname" . }}
```

Comment:
What is the 'template "foo"' doing and how is it different from .Values.... ?

Reply:
a template is basically a function definition. Instead of repeating some code everywhere, you can put that code inside a template and then just call that template wherever you want that definition (in this case the template is defined in `_helpers.tpl`). Is the same that @sbezverk mentioned for the rbac definitions

Here is some helm [documentation](https://github.com/kubernetes/helm/blob/master/docs/chart_template_guide/named_templates.md) about it.